src/main/java/javaserver/handlers/FileHandler.java

```diff
@@ -23,24 +24,17 @@ public FileHandler(String fileUri, ResourceUtility resourceUtility) {
 
     public Response handleRequest(Request request) {
 
-        String method = request.getMethod();
-        if (!method.equals("GET")) {
+        if (!isGetRequest(request)) {
             response.setStatus(Status.MethodNotAllowed);
             return response;
         }
 
-        if (isPartialRequest(request)) {
-            response = new PartialHandler(fileUri, resourceUtility).handleRequest(request);
-            return response;
-        }
-
-        if (isImageRequest(request)) {
-            response = new ImageHandler(fileUri, resourceUtility).handleRequest(request);
-            return response;
+        Handler fileHandler = getTypeOfFileHandler(request);
```

Comment:
I meant that another class should have the responsibility of doing this. Maybe a FileHandlerDeterminer? I know this class is still relatively small but in real world cases, it would be more maintainable that way.